r/R/parquet.R

```diff
@@ -15,15 +15,102 @@
 # specific language governing permissions and limitations
 # under the License.
 
+#' @include R6.R
+
+`parquet::arrow::FileReader` <- R6Class("parquet::arrow::FileReader",
+  inherit = `arrow::Object`,
+  public = list(
+    ReadTable = function(col_select = NULL) {
+      col_select <- enquo(col_select)
+      if(quo_is_null(col_select)) {
+        shared_ptr(`arrow::Table`, parquet___arrow___FileReader__ReadTable1(self))
+      } else {
+        all_vars <- shared_ptr(`arrow::Schema`, parquet___arrow___FileReader__GetSchema1(self))$names
+        indices <- match(vars_select(all_vars, !!col_select), all_vars) - 1L
+        shared_ptr(`arrow::Table`, parquet___arrow___FileReader__ReadTable2(self, indices))
+      }
+    },
+    GetSchema = function(column_indices = NULL) {
+      if (is.null(column_indices)) {
+        shared_ptr(`arrow::Schema`, parquet___arrow___FileReader__GetSchema1(self))
+      } else {
+        shared_ptr(`arrow::Schema`, parquet___arrow___FileReader__GetSchema2(self, column_indices))
+      }
+
+    }
+  )
+)
+
+`parquet::arrow::ArrowReaderProperties` <- R6Class("parquet::arrow::ArrowReaderProperties",
+  inherit = `arrow::Object`,
+  public = list(
+    read_dictionary = function(column_index) {
+      parquet___arrow___ArrowReaderProperties__get_read_dictionary(self, column_index)
+    },
+    set_read_dictionary = function(column_index, read_dict) {
+      parquet___arrow___ArrowReaderProperties__set_read_dictionary(self, column_index, read_dict)
+    }
+  ),
+  active = list(
+    use_threads = function(use_threads) {
+      if(missing(use_threads)) {
+        parquet___arrow___ArrowReaderProperties__get_use_threads(self)
+      } else {
+        parquet___arrow___ArrowReaderProperties__set_use_threads(self, use_threads)
+      }
+    }
+  )
+)
+
+#' Create a new ArrowReaderProperties instance
+#'
+#' @param use_threads use threads ?
+#'
+#' @export
+parquet_arrow_reader_properties <- function(use_threads = TRUE) {
+  shared_ptr(`parquet::arrow::ArrowReaderProperties`, parquet___arrow___ArrowReaderProperties__Make(isTRUE(use_threads)))
+}
+
+#' Create a FileReader instance
+#'
+#' @param file file
+#' @param props reader file properties, as created by [parquet_arrow_reader_properties()]
```

Comment:
This isn't very helpful documentation--can you elaborate, or at least point to the Arrow docs where they are enumerated?

Reply:
At this point, these are really to check boxes, I suppose we can do a doc sweep at some point

Reply:
Ok, I'll take on doc improvements after this merges.